contracts/extensions/bridge-operator-governance/BridgeManager.sol

```diff
@@ -0,0 +1,503 @@
+   * @return removeds An array of boolean values indicating whether each bridge operator was successfully removed.
+   */
+  function _removeBridgeOperators(
+    address[] memory bridgeOperators
+  ) internal nonDuplicate(bridgeOperators) returns (bool[] memory removeds) {
+    uint256 length = bridgeOperators.length;
+    removeds = new bool[](length);
+    // simply skip remove operations if inputs are empty.
+    if (length == 0) {
+      return removeds;
+    }
+
+    mapping(address => address) storage _governorOf = _getGovernorOf();
+    EnumerableSet.AddressSet storage _governorSet = _getGovernorsSet();
+    EnumerableSet.AddressSet storage _bridgeOperatorSet = _getBridgeOperatorSet();
+    mapping(address => BridgeOperatorInfo) storage _governorToBridgeOperatorInfo = _getGovernorToBridgeOperatorInfo();
+
+    address governor;
+    address bridgeOperator;
+    uint256 accumulatedWeight;
+    BridgeOperatorInfo memory bridgeOperatorInfo;
+
+    for (uint256 i; i < length; ) {
+      bridgeOperator = bridgeOperators[i];
+      governor = _governorOf[bridgeOperator];
+
+      _requireNonZeroAddress(governor);
+      _requireNonZeroAddress(bridgeOperator);
+
+      bridgeOperatorInfo = _governorToBridgeOperatorInfo[governor];
+      if (bridgeOperatorInfo.addr != bridgeOperator) revert ErrInvalidArguments(msg.sig);
+
+      removeds[i] = _bridgeOperatorSet.contains(bridgeOperator) && _governorSet.contains(governor);
+      if (removeds[i]) {
+        _governorSet.remove(governor);
+        _bridgeOperatorSet.remove(bridgeOperator);
+
+        delete _governorOf[bridgeOperator];
+        delete _governorToBridgeOperatorInfo[governor];
+        accumulatedWeight += bridgeOperatorInfo.voteWeight;
+      }
+
+      unchecked {
+        ++i;
+      }
+    }
+
+    TOTAL_WEIGHTS_SLOT.subAssign(accumulatedWeight);
+
+    _notifyRegisters(IBridgeManagerCallback.onBridgeOperatorsRemoved.selector, abi.encode(bridgeOperators, removeds));
+
+    emit BridgeOperatorsRemoved(removeds, bridgeOperators);
+  }
+
+  /**
+   * @dev Sets threshold and returns the old one.
+   *
+   * Emits the `ThresholdUpdated` event.
+   *
+   */
+  function _setThreshold(
+    uint256 numerator,
+    uint256 denominator
+  ) internal virtual returns (uint256 _previousNum, uint256 _previousDenom) {
+    if (numerator > denominator) revert ErrInvalidThreshold(msg.sig);
+
+    _previousNum = NUMERATOR_SLOT.load();
+    _previousDenom = DENOMINATOR_SLOT.load();
+    NUMERATOR_SLOT.store(numerator);
+    DENOMINATOR_SLOT.store(denominator);
+
+    emit ThresholdUpdated(NONCE_SLOT.postIncrement(), numerator, denominator, _previousNum, _previousDenom);
+  }
+
+  function _getBridgeOperators() internal view returns (address[] memory) {
+    return _getBridgeOperatorSet().values();
+  }
+
+  function _getGovernors() internal view returns (address[] memory) {
+    return _getGovernorsSet().values();
+  }
```

Comment:
Why should we create new methods just for getting inner values?

Reply:
we prefer using `internal` and `external` methods over using `public` methods.